metabase/metabase
# Add basic dataset metadata editor UI

Adds basic dataset metadata editing UI on `/dataset/:id/metadata`

**Out of scope**

Metadata editing requires many changes, so the work is going to be broken down into a few PRs targeting `dataset/metadata-editor` feature branch. This PR is lacking a few things:

* API integration: it's not possible to save metadata changes
* field mapping UI for native datasets (for native datasets, people will need to explicitly set to which DB column a field is mapped to)
* semantic type input (this PR uses a simple "select" input, though we actually want to have a slightly customized AccordionList thing
* field formatting UI (e.g. once you set the field's semantic type to "Cost", you should be able to specify its currency, number formatting options, etc.)

### To Verify

1. Open a dataset with a structured query
2. Click the dataset name to open the details sidebar and click the new "Customize metadata" button
3. You should end up on the dataset editor page and:
    * you should see navigation tabs in the top bar, the "Metadata" tab should be selected
    * you should not see the query editor, only the results table and a sidebar on the right
    * the right sidebar should display field metadata: display name, field description, semantic type, etc.
    * you shouldn't be able to resize the results table
4. The sidebar should display the first field's metadata by default
5. Click on a few column names, they should get selected and the sidebar should now display their metadata
6. Ensure metadata matches the one on `/admin/datamodel`
7. Click the "Query" tab in the top bar, the editor should switch into the query editor mode. The metadata sidebar should disappear, and the query editor should work as described in #19103
7. Open the metadata editor for a native dataset. It should look pretty much the same, but the fields should be lacking most of the metadata (it's unclear from which table a column comes from, so we can't "inherit" the metadata by default and people will need to explicitly set to which real column a field is mapped to)

### Demo

https://user-images.githubusercontent.com/17258145/144880535-31eb5be3-67f9-4b41-b7e6-ca199f058be6.mov
